### ISSUE TITLE
docs: MultiStartProdigy as the default search in start_here.py

### ISSUE THREAD

**Issue by Jammy2211** [Jammy2211/autolens_workspace#366]
## Overview

Every dataset type in `autolens_workspace` and `autogalaxy_workspace` has a `start_here.py` — the flagship entry-point a new user runs first. All 13 currently fit with `af.Nautilus`, a nested sampler taking 10-30 minutes that returns a full posterior. `af.MultiStartProdigy` is the learning-rate-free JAX multi-start gradient MAP optimizer (GIGA-Lens multi-start recipe, arXiv:2202.07663; Prodigy update rule, arXiv:2306.06101) and reaches the same maximum far faster.

Make the first fit a user runs **fast**, be explicit that a MAP optimizer returns a single best-fit model with **no posterior and no parameter errors**, and point at the folder's `modeling.py` as the example that runs Nautilus for the full posterior. `modeling.py` keeps Nautilus and gains only 2-3 lines of prose — **no code**.

## Plan

- Convert 8 of 13 `start_here.py` files to `af.MultiStartProdigy`, replacing the `af.Nautilus` block outright.
- Add a `__Posterior__` prose block to each: multi-start gradient optimizers return a single MAP point estimate, no errors, no covariances — run `modeling.py` for the full posterior.
- Leave 5 cells on `af.Nautilus` (three independent, confirmed blockers — see below) with 2-3 lines explaining why the fast MAP default is not available for that data type.
- Add the 2-3 line `MultiStartProdigy` mention to the Search docstring of all 13 `modeling.py` files, with **zero** code changes there.
- Regenerate notebooks for both workspaces; verify with `check_sizes.sh` + the smoke runner.

<details>
<summary>Detailed implementation plan</summary>

### Affected Repositories
- autolens_workspace (primary)
- autogalaxy_workspace

### Branch Survey
| Repository | Current Branch | Dirty? |
|-----------|---------------|--------|
| ./autolens_workspace | main | clean |
| ./autogalaxy_workspace | main | clean |

**Suggested branch:** `feature/multistart-prodigy-start-here`
(`worktree_check_conflict` run, exit 0 — no conflict. Note `origin/feature/document-prodigy-jax-optimizers` on autolens_workspace is already merged into main; it is the `searches.py` prose.)

### Convert to `af.MultiStartProdigy` (8 files)

All parametric (MGE / Sersic, **no `Pixelization`**) and already `use_jax=True`, so none of the settings the pixelized campaign found load-bearing apply — no `resurrect=True`, no kernel-regularization choice, no extended `n_steps`.

- `autolens_workspace/scripts/{imaging,interferometer,multi_galaxy,group}/start_here.py`
- `autogalaxy_workspace/scripts/{imaging,interferometer,multi_galaxy,cluster}/start_here.py`

```python
search = af.MultiStartProdigy(
    path_prefix=Path("imaging"),   # unchanged per cell
    name="start_here",
    unique_tag=dataset_name,
    n_starts=48,   # independent optimizations run in parallel via jax.vmap
    n_steps=300,   # ceiling; auto-convergence stops early once the best fit plateaus
    iterations_per_quick_update=50,
    live_visual_update=True,   # only where the cell already set it
)
```

`n_starts=48` / `n_steps=300` are the class defaults (`AbstractMultiStartGradient.__init__`). Auto-convergence is on by default, so `n_steps` is a max budget, not a fixed cost.

**Delete `n_batch`** — it is Nautilus's own proposal-count knob that autofit merely forwards, not a `MultiStartProdigy` parameter; its trailing comment goes with it. Keep `iterations_per_quick_update` / `live_visual_update` (both supported via the base class, `abstract_search.py:137`) but retune the cadence: existing values (1000; 250000 in point_source) are Nautilus iteration counts and would never fire inside a 300-step budget.

Prose in the same files: rewrite the `__Model Fit__` "nested sampling algorithm Nautilus" sentence (Nautilus also appears in the `AnalysisImaging` sentence beneath it); add the `__Posterior__` block; revise the runtime estimates downward.

Post-fit result usage in all 8 is already MAP-safe (`result.info`, `result.max_log_likelihood_tracer`, `result.max_log_likelihood_fit`) — verified, no sample/posterior access to fix.

### Keep `af.Nautilus` + a "not available here" note (5 files)

| Cell | Blocker |
|---|---|
| autolens `point_source` | `AnalysisPoint`; point-source JAX gradients do not exist. Enabling work is `PyAutoMind/draft/feature/autolens/point_source_chi_squared_paper_variants_phase_5_jax_gradients.md`, whose stated goal is "so gradient searches (`af.MultiStartProdigy` / Adam-family) work on point-source fits". Blocked on phase 2 merge. |
| autolens `cluster` | Also `AnalysisPoint` (7 source systems over a `FactorGraphModel`), same blocker. Also ends on `aplt.corner_anesthetic(samples=result_list[0].samples)` — a posterior corner plot a MAP optimizer cannot feed. |
| autolens `weak` | `AnalysisWeak` pinned to `use_jax=False` by an in-script comment citing PyAutoLens#614 ("Weak-lensing visualization crashes on JAX path: fit.shear_yx lacks .grid"), confirmed **OPEN**. A gradient search requires `use_jax=True`. |
| autolens `multi` | Multi-band `FactorGraphModel` `value_and_grad` cold compile is **unbounded on CPU** — >2h wall, ~1h inside one XLA compile, cache MISS, vs 117s single-band. Unresolved in `PyAutoMind/draft/research/autofit/multi_band_factorgraphmodel_value_and_grad_cold.md`, whose named reproducer IS this script's 4-band `cosmos_web_ring` fit. |
| autogalaxy `multi` | Same shape — 2 bands at heterogeneous pixel scales (g=0.08, r=0.12), same unresolved compile blocker. |

Smoke would NOT have caught the `multi` case: `PYAUTO_TEST_MODE=2` skips search sampling entirely.

### `modeling.py` (13 files) — prose only, zero code

2-3 lines inside the existing `__Search__` docstring. Three files use `__Search + Analysis__` instead and have no `__Search__` header: autolens `multi_galaxy`, autogalaxy `multi_galaxy`, autogalaxy `cluster`.

- Converted folders: this example uses `Nautilus` for the **full posterior**; the folder's `start_here.py` uses `af.MultiStartProdigy` for a much faster MAP point estimate with no errors.
- Blocked folders: adapt — `MultiStartProdigy` is the fast MAP alternative used in other dataset types' `start_here.py`, not yet available for this data type.

### Key Files
- `autolens_workspace/scripts/guides/modeling/searches.py` (lines 191-256) — canonical long-form `MultiStartProdigy` prose. `start_here.py` gets a condensed version, **not** a duplicate.
- `PyAutoFit/autofit/non_linear/search/mle/multi_start_gradient/search.py` — `MultiStartProdigy` + `AbstractMultiStartGradient.__init__` parameter contract.
- `scripts/check_sizes.sh` (both workspaces) — bulk-edit shrinkage guard; run before committing.

### Brain phase-split override

`bin/pyauto-brain feature` returned `too-large (score 10) → split-into-phases` (design / core_api / workspace_examples / docs). **Overridden.** Brain scores difficulty off repo count; the proposed `core_api` phase is vacuous because no library code is touched at all. This is one uniform prose + search-swap sweep over 26 files in 2 repos = one PR.

### Verification
- `scripts/check_sizes.sh` clean in both workspaces.
- `python .github/scripts/run_smoke.py` from each workspace root — green summary. **Caveat for the PR body:** smoke runs under `PYAUTO_TEST_MODE=2`, which skips search sampling; it proves the scripts compose and run end-to-end, it does **not** exercise a real `MultiStartProdigy` fit.
- Real-fit spot check: run one converted cell without test-mode to confirm the search converges and writes output.
- `git diff --stat` pre-flight for binary output leakage; notebooks regenerated.

</details>

## Original Prompt

<details>
<summary>Click to expand starting prompt</summary>

```markdown
# start_here.py: MultiStartProdigy as the default search across dataset types

Type: docs
Target: autolens_workspace
Repos:
- autolens_workspace
- autogalaxy_workspace
Difficulty: medium
Autonomy: supervised
Priority: normal
Status: draft

## Original request (verbatim)

> Use MultiStartProdigy in start here examples of all dataset types, with
> description of posterior and say modeling.py gives full posterior. In
> modeling.py mention MultiStartProdigy and its use in start_here.py in like 2
> or 3 lines in the Search Docstring but dont include any code. Do this on
> autogalaxy and lens workspaces.

## Intent

Make the flagship `start_here.py` entry-point of every dataset type run the fast
`af.MultiStartProdigy` MAP optimizer instead of `af.Nautilus`, so a new user's
first fit finishes quickly. The prose must be explicit that a multi-start
gradient optimizer returns a **single best-fit (MAP) model with no posterior and
no parameter errors**, and must point at the folder's `modeling.py` as the
example that runs `Nautilus` and yields the full posterior.

`modeling.py` keeps `Nautilus` and keeps its code unchanged — it only gains 2-3
lines of prose in its `__Search__` (or `__Search + Analysis__`) docstring
mentioning `MultiStartProdigy` and its use in `start_here.py`. **No code is
added to `modeling.py`.**

## Scope — 13 dataset-type cells, 8 converted

All 8 convertible cells are parametric (MGE / Sersic, no `Pixelization`) and
already set `use_jax=True`, so they need no `resurrect=True`, no
kernel-regularization choice and no extended `n_steps` — the settings that the
pixelized campaign found load-bearing do not apply here.

Convert to `af.MultiStartProdigy` (replacing the `af.Nautilus` block outright —
no commented-out Nautilus left behind):

- autolens_workspace: `imaging`, `interferometer`, `multi_galaxy`, `group`
- autogalaxy_workspace: `imaging`, `interferometer`, `multi_galaxy`, `cluster`

Delete `n_batch` when converting — it is Nautilus's own proposal-count knob that
autofit merely forwards, not a `MultiStartProdigy` parameter. Keep
`iterations_per_quick_update` / `live_visual_update` (supported via the base
class) but retune the cadence: existing values are Nautilus iteration counts and
would never fire inside a 300-step budget.

## Blocked cells — keep Nautilus, explain the gap (human decisions, 2026-07-29)

Five cells cannot run a gradient search today. They **keep `af.Nautilus`** and
instead gain 2-3 lines of prose noting that `MultiStartProdigy` is the fast MAP
default in the other dataset types but is not yet available for this data type,
with the reason:

- `point_source` (`AnalysisPoint`) — point-source JAX gradients do not exist
  yet. Enabling work is
  `draft/feature/autolens/point_source_chi_squared_paper_variants_phase_5_jax_gradients.md`,
  whose stated goal is "so gradient searches (`af.MultiStartProdigy` /
  Adam-family) work on point-source fits". Still blocked on phase 2 merge.
- `cluster` — also `AnalysisPoint` (7 source systems over a `FactorGraphModel`),
  same blocker. Note it additionally ends on
  `aplt.corner_anesthetic(samples=result_list[0].samples)`, a posterior corner
  plot a MAP optimizer could not feed.
- `weak` (`AnalysisWeak`) — pinned to `use_jax=False` by an in-script comment
  citing PyAutoLens#614 ("Weak-lensing visualization crashes on JAX path:
  fit.shear_yx lacks .grid"), confirmed OPEN on 2026-07-29. A gradient search
  requires `use_jax=True`.
- `multi` (autolens) and `multi` (autogalaxy) — multi-band `FactorGraphModel`
  `value_and_grad` cold compile is **unbounded on CPU** (>2h wall observed, ~1h
  inside one XLA compile, cache MISS, against the 117s single-band figure).
  Recorded unresolved in
  `draft/research/autofit/multi_band_factorgraphmodel_value_and_grad_cold.md`,
  whose named reproducer IS autolens `multi/start_here.py`'s 4-band
  `cosmos_web_ring` fit. autogalaxy `multi` is the same shape (2 bands at
  heterogeneous pixel scales, g=0.08 / r=0.12). Converting these would make the
  flagship multi-wavelength entry point hang for hours on CPU — the opposite of
  the intent. Note smoke would NOT catch it: `PYAUTO_TEST_MODE=2` skips search
  sampling entirely.

Do not write `af.MultiStartProdigy` into these five — it would ship five broken
or unusable entry-point scripts.

## Brain phase-split override (recorded)

`bin/pyauto-brain feature` returned `too-large (score 10) → split-into-phases`
(design / core_api / workspace_examples / docs). **Overridden.** Brain scores
difficulty off repo count; the proposed `core_api` phase is vacuous because no
library code is touched at all. This is one uniform prose + search-swap sweep
over 26 files in 2 repos = one PR.

## Reference prose

`autolens_workspace/scripts/guides/modeling/searches.py` (the
`__MultiStartProdigy (JAX multi-start gradient optimizers)__` section) is the
canonical long-form description — GIGA-Lens multi-start provenance
(arXiv:2202.07663), Prodigy's learning-rate-free rule (arXiv:2306.06101), and
the "returns a single best-fit model, not a posterior with errors, so Nautilus
remains the default when uncertainties are required" framing. `start_here.py`
prose should be a short version of this, not a duplicate of it.

## Acceptance

- 8 `start_here.py` files run `af.MultiStartProdigy`; posterior caveat +
  pointer to `modeling.py` present in each.
- 13 `modeling.py` files carry the 2-3 line `MultiStartProdigy` mention in their
  Search docstring, and **no** new code.
- 5 blocked `start_here.py` files carry the "not yet available here" note and
  still run `af.Nautilus`.
- Smoke suite green in both workspaces; notebooks regenerated.
- `scripts/check_sizes.sh` clean (bulk-edit guard).
```

</details>

**Comment by Jammy2211** [comment]
## Shipped — both PRs open

- **autolens_workspace** → https://github.com/PyAutoLabs/autolens_workspace/pull/373
- **autogalaxy_workspace** → https://github.com/PyAutoLabs/autogalaxy_workspace/pull/180

Both carry `pending-release`.

### What landed

**8 `start_here.py` converted** to `af.MultiStartProdigy`, each with a `__Multi Start Gradient Optimization__` section and a `__Posterior__` section saying plainly that a MAP optimizer gives a single best-fit model with no posterior/errors/covariances, pointing at the folder's `modeling.py` for the full Nautilus posterior:

- autolens: `imaging`, `interferometer`, `multi_galaxy`, `group`
- autogalaxy: `imaging`, `interferometer`, `multi_galaxy`, `cluster`

**5 left on `af.Nautilus`** with a `__Why Not MultiStartProdigy?__` note: autolens `point_source`, `cluster`, `weak`, `multi` and autogalaxy `multi`. Reasons are in the PR bodies — non-differentiable `AnalysisPoint` lens-equation solve, PyAutoLens#614 forcing `use_jax=False`, and the unbounded multi-band gradient compile.

**13 `modeling.py`** gained a 2-3 line Search-docstring mention with **zero code changes** (verified by filtering the diff for Python statements — none).

### Scope changes from the original request

Two blockers surfaced during scoping that reduced the conversion from 13 cells to 8, both confirmed against existing records rather than assumed:

1. Point-source JAX gradients do not exist yet (the Phase-5 prompt that would enable them is still blocked), and `weak` is pinned to `use_jax=False` by the still-OPEN PyAutoLens#614.
2. The multi-band `FactorGraphModel` `value_and_grad` cold compile is unbounded on CPU (>2h observed vs 117s single-band), and its recorded reproducer is literally `autolens multi/start_here.py`'s 4-band `cosmos_web_ring` fit. Converting it would have made the flagship multi-wavelength example hang for hours on CPU — and smoke would not have caught it, since `PYAUTO_TEST_MODE=2` skips search sampling.

### Validation

- autolens smoke **16/16**, autogalaxy smoke **12/12**.
- `check_sizes.sh` OK in both repos.
- Real no-test-mode fit of autogalaxy `imaging/start_here.py`: converged early at **step 132** of the 300 ceiling, 7.5 min CPU, max log likelihood −808.19.
- Heart was YELLOW on 4 pre-existing unrelated reasons (validation run dated 2026-07-21; two manifest-drift items from another session's tenant-firewall work); acknowledged by the maintainer and recorded in `active.md`.

### Two API corrections made while converting

- `n_batch` removed — it is Nautilus's proposal-count knob that autofit forwards, not a `MultiStartProdigy` parameter.
- `iterations_per_quick_update` 1000/10000 → 50 — those were Nautilus *iteration* counts and would never have fired inside a 300-**step** gradient budget, silently suppressing the quick-update visualization.
